### PR TITLE
Updates for change log2021.1

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
None

### Summary of the issue:
This PR aims to gather issues found in the beta stage.
The following issue have been found:
1. There is a line in the change log for CLDR 38.1 update, but not for CLDR 39.0 update
2. The blurb indicates that add-ons will need to have their manifest updated. This seems very devleopper oriented, not suitable for a blurb.

### Description of how this pull request fixes the issue:

Modified the change log as follows:
1. Indicate update to CLDR 39.0 rather than 38.1; but referenced both PRs (the one for 38.1 update and the one for 39.0 update)
2. I have rewritten the blurb regarding add-ons to make it more end-user oriented. (Feel free to correct me, English is not may native language.) I have also put all the modifications in the first paragraph and put the note regarding the add-ons in a separate paragraph below.

### Testing strategy:

Check the change log generated by appVeyor.

### Known issues with pull request:
None

### Change log entries:
None of course!

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [ ] Manual testing.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

**Comment by CyrilleB79** [comment]
Passing this PR as ready.
Please double check the following sentence:
> Add-ons will need to be re-tested and authors will need to provide a new version certifying its compatibility with this release. 

This seems grammatically incorrect (singular vs. plural...) to me but I would prefer to have the view of a native English speaker before modifying it. Thanks.

File: user_docs/en/changes.t2t
Comment (seanbudd) [inline user_docs/en/changes.t2t:14]:
```suggestion
Add-ons will need to be re-tested.
Add-on authors will need to release a new version with an updated manifest to certify compatibility with this release.
```

**Review by seanbudd — COMMENTED** [review]
I agree that it would be good to rephrase this. I've added my suggestion to break up the long sentence and reduce ambiguity. Going to await further discussion from others.

File: user_docs/en/changes.t2t
Comment (feerrenrut) [inline user_docs/en/changes.t2t:11]: I think the intention here was to make this an item of the "Note", perhaps @Qchristensen can clarify?

File: user_docs/en/changes.t2t
Comment (feerrenrut) [reply user_docs/en/changes.t2t:14]: This is tricky, as a message to add-on authors this isn't detailed enough. It could say consult the "changes for developers" section and check if your addon relies on any of the changed / moved / removed code. However, for regular users that is way too technical for a release blurb / highlight.

Perhaps we could just add "consult the changes for developers".

File: user_docs/en/changes.t2t
Comment (seanbudd) [reply user_docs/en/changes.t2t:11]: Could we bolden this text and maybe say "Warning" instead. The increase in urgency might decrease the number of people who miss/ignore it.

File: user_docs/en/changes.t2t
Comment (seanbudd) [reply user_docs/en/changes.t2t:14]: Thoughts on the following?

```suggestion
Add-ons will need to be re-tested and release a new version which certifies compatibility with this version of NVDA.
Add-on authors should consult the changes for developers.
```

**Comment by XLTechie** [comment]
Cyrille Bougot wrote:

> Please double check the following sentence:
>       Add-ons will need to be re-tested and authors will need to provide a new version certifying its compatibility with this release.

Add-ons will need to be re-tested and authors will need to provide new 
versions certifying their compatibility with this release.

Do note, that the manifest language already existed in two places in 
changes (lines 29 and 89).

**Comment by XLTechie** [comment]
Reef Turner wrote:

> This is tricky, as a message to add-on authors this isn't detailed enough. It could say consult the "changes for developers" section and check if your addon
> relies on any of the changed / moved / removed code. However, for regular users that is way too technical for a release blurb / highlight.
> 
> Perhaps we could just add "consult the changes for developers".

Is add-on authors not knowing what to do when the API breaks, actually a 
problem that exists?

Not to be contentious, this just seems unnecessarily hand-holdy.

I don't object to the "warning", the bolding, whatever, it just never 
occurred to me that someone writing code intimately linked to a piece of 
software, wouldn't already know where to look when the API they use 
breaks.

**Comment by seanbudd** [comment]
@XLTechie @feerrenrut, my intention (and I think @CyrilleB79's) is to make the message more clear to end-users, rather than developers, that add-ons are breaking if you update. And that you need to wait for devs to fix them and release a new version.

**Comment by feerrenrut** [comment]
The change log is also intended for addon developers. Some of which may not closely follow the development of NVDA. 

I don't think it is unreasonable to provide extra guidance, especially if low cost. In other channels, this guidance seems necessary. This process is relatively new (this is only the second API compatibility change) and we don't have anything explicitly pointing authors to the change log. In other software, especially software libraries, update guides are common.

File: user_docs/en/changes.t2t
Comment (CyrilleB79) [reply user_docs/en/changes.t2t:11]: Looking at the commits in the initial PR (#12376), we can see that all the sentences of the blurb were in the initial paragraph. Then the paragraph was split in two, probably unintentionally when changing "mathematic" to "mathematical".
Putting all the sentences in the first paragraph and the note regarding add-ons in a second one was my initiative in this PR only (not yet merged).

File: user_docs/en/changes.t2t
Comment (CyrilleB79) [reply user_docs/en/changes.t2t:14]: To clarify my intention when rewording this sentence:
1. I have assumed that the blurb was for regular users and that add-on developers know that they need to look at the change for developers. Thus,
2. I absolutely wanted to remove the word manifest in the blurb since
  * the regular user does not now what it is about
  * or much worse the regular user may be tempted to modify himself the manifest in the place of the add-on author, what is not recommended at all; we can see regularly such messages on the mailing lists

Regarding the the following sentence:
> Add-on authors should consult the changes for developers.
I personally do not think it is required since the blurb is already there as an incitation for people to read the rest of the change log. Let me know if you want me to add it anyway.

Comment (CyrilleB79) [reply user_docs/en/changes.t2t:14]: Here is a suggestion. Feel free to comment:
Even if it targets add-on authors, regular users should understand that they need to wait for an updated add-on release.

```suggestion
Add-on authors will need to re-tested their add-on and authors will need to provide a new version certifying its compatibility with this release.

```

**Comment by CyrilleB79** [comment]
@XLTechie wrote:
> Cyrille Bougot wrote:
> > Please double check the following sentence: Add-ons will need to be re-tested and authors will need to provide a new version certifying its compatibility with this release.
> > Add-ons will need to be re-tested and authors will need to provide new versions certifying their compatibility with this release.
> Do note, that the manifest language already existed in two places in changes (lines 29 and 89).

Talking about manifest in the change log is not a problem for the already two mentions:
1. The following sentence is in the dev changes, so no problem to mention menifest:
> Add-on authors are now able to provide NVDA version compatibility information in add-on manifests. (#6275, #9055)
2. The following sentence explains the consequence of the presence of the manifest. It is in a change log item, not in the blurb. And it deals with NVDA application manifest, not add-on ones.
> The NVDA application now includes a manifest, which means that it no longer runs in compatibility mode in Windows Vista.

As explained in my comment in the code, putting the word "manifest" in the blurb seems to me to make it something very common with the risk that regular users twick it to allow compatibility.

**Comment by XLTechie** [comment]
Cyrille Bougot wrote:

> Talking about manifest in the change log is not a problem for the already two mentions:
>
>  1. The following sentence is in the dev changes, so no problem to mention menifest:
>
>       Add-on authors are now able to provide NVDA version compatibility information in add-on manifests. (#6275, #9055)

That is an old item, I'm not sure of its relevance.

I was talking about the first line under changes for developers, in the 
2021.1 beta changes.t2t.

Note: this is an Add-on API compatibility breaking release. Add-ons will 
need to be re-tested and have their manifest updated.

It is a similar note to the one you are trying to remove from the blurb.

I think that since it appears there, it does not need to be emphasized in 
the blurb.

> As explained in my comment in the code, putting the word "manifest" in the blurb seems to me to make it something very common with the risk that regular
> users twick it to allow compatibility.

Yes, I agree with you there.

**Comment by CyrilleB79** [comment]
@XLTechie wrote:
> Yes, I agree with you there.

Oh sorry. I have misunderstood your previous comment.

File: user_docs/en/changes.t2t
Comment (feerrenrut) [reply user_docs/en/changes.t2t:14]:
```suggestion
Add-ons will need to re-tested.
Add-on authors will need to provide a new version certifying its compatibility with this release.
Add-on authors should use the "changes for developers" to guide their update.
```

Comment (CyrilleB79) [reply user_docs/en/changes.t2t:14]: Still refining the formulation.
From @feerrenrut's suggestion:
* added the missing word "be"
* replaced the second occurrence of "Add-on authors" with "They"
Hope it's OK now!

```suggestion
Add-ons will need to be re-tested.
Add-on authors will need to provide a new version certifying its compatibility with this release.
They should use the "changes for developers" to guide their update.

```

Comment (XLTechie) [reply user_docs/en/changes.t2t:14]: Whatever is done here, please get the plurals right. If saying add-ons, or add-on authors, it is awkward to then talk about the add-on in the singular ("it", "a new version").

```suggestion
Add-on authors will need to test their add-ons with this release, and provide new versions certifying their compatibility.
Authors should see the "Changes for Developers" section of the changes file for details.
```

Alternative suggestion:

```suggestion
Add-on authors will need to test their add-ons with 2021.1, and release new add-on versions which declare compatibility with it.
Authors should see the "Changes for Developers" section of the changes file for details.
```

Comment (CyrilleB79) [reply user_docs/en/changes.t2t:14]: There are many comments for these sentence!

Personally, I would be willing to accept @XLTechie's first suggestion, i.e.: 
> Add-on authors will need to test their add-ons with this release, and provide new versions certifying their compatibility.
> Authors should see the "Changes for Developers" section of the changes file for details.
especially for the first sentence.

I prefer this formulation for the following points:
* It indicates clearly who is in charge of add-on re-testing
* It indicates clearly the nature and the objective of the tests, i.e. check and certify compatibility with this release.
* No mention of the manifest in the blurb to avoid the temptation for regular users to modify it
* No singular / plural mismatch
* Reference to the "Changes for dev" section; from the last comments, it seemed a consensus to have this reference.

Please for anybody, if you make an alternative suggestion, indicate your motivation with the points above and new ones. Thanks.

Comment (Qchristensen) [reply user_docs/en/changes.t2t:14]: Since the last time an update broke add-ons was 2019.3, I went back to the blurb for that update, which says simply:

(Point about update to Python and speech subsystem)... Although these changes do break compatibility with older NVDA add-ons, the upgrade to Python 3 ... (highlighting importance of updates).

Working from that, perhaps simply:

Note: This release breaks compatibility with existing add-ons
```suggestion
Note: This release breaks compatibility with existing add-ons

```
I contemplated whether to make it "Note: This release breaks compatibility with existing add-ons, which will need to be updated.", but I'm not convinced that latter part needs to be there - existing add-ons are broken, therefore they need to be updated.


As noted.  Add-on authors should be familiar with reading the changes for developers section.  I don't think we should need to specify that they read that section.  Since we are already saying it breaks add-ons, it should follow that the extra detail on why should be contained in that changes for developers section.

Comment (CyrilleB79) [reply user_docs/en/changes.t2t:14]: I think the idea when telling that add-on authors should provide new versions of their add-ons is to avoid questions or issues such as #12426. So IMO specifying it remains important.

Anyway, I think everyone has been able to write their arguments on how to write this sentence. At the end, it's up to NVAccess people to decide.
@feerrenrut, @seanbudd, @Qchristensen for now there is no converging proposal from you three. Could you try to agree on what is important (re-reading the whole thread) or indicate who of you three is in charge to decide/validate at the end? Thanks.

**Comment by XLTechie** [comment]
#12426 was not written by an add-on author, so it would not have been the 
target of the notice anyway.

While I like my suggestion if we have to go with the wordy version, I 
personally prefer what @Qchristensen wrote--it is more inline with my 
initial thoughts, for what my opinion is worth.

**Comment by CyrilleB79** [comment]
I guess with the downgrade of eSpeak (#12495), a new beta will be released. Could we progress this PR as well to have it merged and translated?

All NVAccess people (@Qchristensen , @seanbudd , @feerrenrut), your replies are pending on the following threads:
* all: in the thread regarding add-on breaking release note (comment on line 14): A converging point of view among NVAccess people (or indicating who is making the final decision) is required. The pending question is if a shorter or a more detailed sentence/paragraph should be included regarding this point.
* @Qchristensen: in the thread regarding Flash withdrawal (comment on line 11): Indicate if the initial intention was to put Flash withdrawal in the furst paragraph of the blurb or it should stand in the note with add-on breaking compa release. IMO first alternative was intended and this point can be closed without changing anything.
Thanks.

File: user_docs/en/changes.t2t
Comment (feerrenrut) [reply user_docs/en/changes.t2t:14]: Yep, let's go with @Qchristensen's suggestion. However, I think we should use a list to have discreet items for the "Note", including the flash item.

File: user_docs/en/changes.t2t
Comment (feerrenrut) [inline user_docs/en/changes.t2t:15]:
```suggestion

Note:
 - This release breaks compatibility with existing add-ons.
 - This release also drops support for Adobe Flash.

```

Comment (Qchristensen) [reply user_docs/en/changes.t2t:15]: Thanks @feerrenrut - I'm in favour of that wording.

Comment (seanbudd) [reply user_docs/en/changes.t2t:15]: Thanks @feerrenrut, this wording looks good.

**Review by feerrenrut — DISMISSED** [review]

**Comment by feerrenrut** [comment]
Urrgh, fixing conflicts using the web editor has changed all the line endings. I'll fix this up tomorrow.

**Review by Qchristensen — DISMISSED** [review]
Looks good, thanks Reef!

**Pull request merged**
